content/webapp/pages/pages/[pageId].tsx

```diff
@@ -115,15 +116,19 @@ function getFeaturedPictureWithTasl(
 export const getServerSideProps: GetServerSideProps<Props | AppErrorProps> =
   async context => {
     const serverData = await getServerData(context);
-    const { id } = context.query;
+    const { pageId } = context.query;
 
     const client = createClient(context);
 
-    const vanityUrl = isVanityUrl(id as string, context.resolvedUrl)
+    if (!looksLikePrismicId(pageId)) {
+      return { notFound: true };
+    }
```

Comment:
This is logic that used to be duplicated in the routing layer; we've gradually migrated to putting it in `getServerSideProps` because we can lean on the type system – among other things, this method tells us that `pageId` is a `string` if it passes.